### PR TITLE
check install setup using virtualenv

### PR DESCRIPTION
**Pull request by tejalwakchoure**
## What? Why?
Fix # 119 check install setup using virtualenv 

Changes proposed in this pull request:
- Added a bash script (install_package_dependencies_for_virtualenv.sh) containing installation setup for virtualenv

## Checks
- [ ] Single commit and [No merge commits](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [x] Make sure you have added required documentation
- [ ] If you have refactored a code, make sure that you have compared the outputs pre and post refactoring. Add both pre and post refaoring screenshots below if the output is visual

## Notify reviewers
@prasadtalasila

### CODE REVIEW

**Comment by prasadtalasila** [comment]
@tejalwakchoure Unless I am grossly mistaken, I don't see any virtualenv setup in the script. I also see too many pip installations; why are they needed? I mean, if they are needed in the first place, they would be in the requirements.txt, right?

**Comment by tejalwakchoure** [comment]
@prasadtalasila I have added the virtualenv setup and removed the duplicate pip installations.  Everything works fine with the new script

File: etc/install_package_dependencies_for_virtualenv.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+PROJECT_PATH=$(pwd)
+
+# Install Python-PIP
+sudo pip3 install --upgrade pip
+
+# Install virtualenv, create a virtual environment and activate it
+sudo pip3 install virtualenv
+virtualenv env
+source env/bin/activate
```
Comment (prasadtalasila) [inline etc/install_package_dependencies_for_virtualenv.sh:12]: @tejalwakchoure Ideally, the installation file would have commands upto this point. After this we should be able to just run 
```
sudo -H pip3 install -r requirements.txt
```
to install all the required dependencies.

With virtualenv option, I see a need to remove the above command from [install_package_dependencies.sh](https://github.com/DeveloperCAP/MLCAT/blob/master/etc/install_package_dependencies.sh). 

One way to think about installation is:
```
install_os_packages.sh     #performs deb / direct package installations
install_virtualenv.sh           #runs required commands for setting up virtualenv
install.sh                             #performs direct installation of pip packages to base OS
```

@tejalwakchoure you can take the help of @AakankshaSanctis in order to put the necessary scripts in place.

**Review by prasadtalasila — CHANGES_REQUESTED** [review]

**Comment by prasadtalasila** [comment]
Also the travis build is not happening. Does anyone know the reason for the same?

File: etc/install_package_dependencies_for_virtualenv.sh (hunk 1 above)
Comment (hrishikesh-dahiya) [reply etc/install_package_dependencies_for_virtualenv.sh:12]: I think separating all the installation packages won't be necessary, as virtualenv is preferred over base OS installation, better to change the installations to virtualenv and have a simpler installation process

**Comment by hrishikesh-dahiya** [comment]
The travis-ci build is now running.

**Comment by prasadtalasila** [comment]
Also our existing install script is not downloading the NLTK wordnet. Please add the following command to the install script.
```
# install nltk corpus wordnet
python -m nltk.downloader wordnet
```

**Comment by hrishikesh-dahiya** [comment]
`deactivate` at end of the shell script so that user exits the virtualenv at the end of installation process and transfer the changes to `install_package_dependencies.sh`, we don't need both shell scripts, just for virtualenv. Remove the `install_package_dependencies_for_virtualenv.sh` . Also, squash the commits to one.

**Comment by AakankshaSanctis** [comment]
Should the virtualenv installation include install_package_dependencies or just the ones in requirements.txt?Is install_package_dependencies.sh to be done separately outside the virtualenv?

**Comment by hrishikesh-dahiya** [comment]
when run, `install_package_dependencies.sh` should install packages in virtualenv which also includes `requirements.txt` installations

**Comment by AakankshaSanctis** [comment]
So,basically we should have one script which installs package dependencies and requirements in the virtual environment and put this all together in install_vitualenv.sh

**Comment by hrishikesh-dahiya** [comment]
yes, you can keep the previous name `install_package_dependencies.sh` also, just update the script with `deactivate` at the end so that user gets out of the virtualenv and squash to single commit
 
A separate wiki page can be created after it to specify the instructions for installation and running the tool.

**Comment by tejalwakchoure** [comment]
If we need the python package dependencies to be installed only in the virtual environment, why do we need to install them separately in install_package_dependencies.sh considering most of them (numpy, scipy, etc.) are already installed when we run requirements.txt?

**Comment by hrishikesh-dahiya** [comment]
afaik, the shell script runs the requirements.txt in the root directory, there is a last command in the script call pip install -r requirements.txt, we don't run requirements.txt separately

please refer this https://github.com/DeveloperCAP/MLCAT/blob/master/etc/install_package_dependencies.sh

**Comment by tejalwakchoure** [comment]
No I meant when we run pip install -r requirements.txt we anyway install most of these packages. Then why do we need sudo apt-get install python3-xyz

**Comment by hrishikesh-dahiya** [comment]
i get your point, try removing the redundant installs and run them using requirements.txt, if it works, we can remove all the unnecessary stuff there

File: etc/install_package_dependencies_for_virtualenv.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+cd Infomap
+wget http://www.mapequation.org/downloads/Infomap.zip
+unzip Infomap.zip
+make
+cd examples/python
+make python3
+python3 example-networkx.py
+
+# For installation(pip) Python-iGraph
+pip3 install igraph
```
Comment (hrishikesh-dahiya) [inline etc/install_package_dependencies_for_virtualenv.sh:33]: try removing `pip3 install igraph` also, igraph install may work without this

**Pull request merged**